File: serializer-deserializer/src/test/java/com/amazonaws/services/schemaregistry/serializers/protobuf/ProtobufGenerator.java

package com.amazonaws.services.schemaregistry.serializers.protobuf;

import Foo.Contact;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax2.Basic;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax2.ComplexNestingSyntax2;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax2.alltypes.AllTypesSyntax2;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax2.basic.BasicSyntax2;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax2.basic.ProtodevelaslProtoProtoProtodevelBar3;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax2.snake_case.SnakeCaseFile;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.ComplexNestingSyntax3;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.alltypes.AllTypes;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.alltypes.AnEnum;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.alltypes.AnotherTopLevelMessage;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.basic.ConflictingNameOuterClass;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.basic.NestedConflictingClassNameOuterClass;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.basic.Basicsyntax3;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.basic.Foo1;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.basic.HyphenAtedProtoFile;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.basic.Unicode;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.multiplefiles.A;
import com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.snake_case.AnotherSnakeCaseProtoFile;
import com.amazonaws.services.schemaregistry.utils.apicurio.FileDescriptorUtils;
import com.amazonaws.services.schemaregistry.utils.apicurio.syntax2.WellKnownTypesTestSyntax2;
import com.amazonaws.services.schemaregistry.utils.apicurio.syntax3.WellKnownTypesTestSyntax3;
import com.google.common.collect.ImmutableList;
import com.google.protobuf.*;
import com.google.protobuf.Enum;
import com.google.type.*;
import lombok.SneakyThrows;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;
import java.util.stream.Stream;

/**
 * Generates Protobuf objects to be used during testing
 */
public class ProtobufGenerator {
    public static List<Message> getAllPOJOMessages() {
        return Stream.of(
            BASIC_REFERENCING_MESSAGE,
            BASIC_SYNTAX2_MESSAGE,
            BASIC_SYNTAX3_MESSAGE,
            NESTING_MESSAGE_PROTO2,
            NESTING_MESSAGE_PROTO3,
            NESTING_MESSAGE_PROTO3_MULTIPLE_FILES,
            JAVA_OUTER_CLASS_WITH_MULTIPLE_FILES_MESSAGE,
            JAVA_OUTER_CLASS_MESSAGE,
            UNICODE_MESSAGE,
            NESTED_CONFLICTING_NAME_MESSAGE,
            ALL_TYPES_MESSAGE_SYNTAX3,
            ALL_TYPES_MESSAGE_SYNTAX2,
            WELL_KNOWN_TYPES_SYNTAX_2,
            WELL_KNOWN_TYPES_SYNTAX_3
        ).collect(Collectors.toList());
    }

    public static List<DynamicMessage> getAllDynamicMessages() {
        return Stream.of(
            BASIC_REFERENCING_DYNAMIC_MESSAGE,
            createDynamicProtobufRecord(),
            createDynamicNRecord(),
            createDynamicMessageFromPOJO(ALL_TYPES_MESSAGE_SYNTAX2),
            createDynamicMessageFromPOJO(ALL_TYPES_MESSAGE_SYNTAX3),
            createDynamicMessageFromPOJO(WELL_KNOWN_TYPES_SYNTAX_2),
            createDynamicMessageFromPOJO(WELL_KNOWN_TYPES_SYNTAX_3)
            //Add all types,
        ).collect(Collectors.toList());
    }

    @SneakyThrows
    private static DynamicMessage createDynamicMessageFromPOJO(Message pojo) {
        byte[] pojoBytes = pojo.toByteArray();
        return DynamicMessage
            .newBuilder(pojo.getDescriptorForType())
            .mergeFrom(pojoBytes)
            .build();
    }

    public static Basic.Address createCompiledProtobufRecord() {
        return Basic.Address.newBuilder()
                .setStreet("410 Terry Ave. North")
                .setCity("Seattle")
                .setZip(98109)
                .build();
    }

    public static DynamicMessage createDynamicProtobufRecord() {
        List<Descriptors.FieldDescriptor> fieldDescriptorList = Basic.Address.getDescriptor().getFields();
        return DynamicMessage.newBuilder(Basic.Address.getDescriptor())
                .setField(fieldDescriptorList.get(0), "5432 82nd St")
                .setField(fieldDescriptorList.get(1), 123456)
                .setField(fieldDescriptorList.get(2),"Seattle")
                .build();
    }

    public static DynamicMessage createDynamicNRecord() {
        return DynamicMessage.newBuilder(ComplexNestingSyntax3.N.getDescriptor())
                .setField(ComplexNestingSyntax3.N.getDescriptor().findFieldByName("A"), 100)
                .build();
    }

    /**
     * Creates a Message from a dynamic schema that is only compiled during runtime.
     * There are no POJOs pre-compiled for this schema.
     */
    public static Message createRuntimeCompiledRecord() throws Descriptors.DescriptorValidationException {
        String nonPojoExistentSchemaDefinition =
            "package foo; message NonExistentSchema { optional string a = 1; }";

        Descriptors.FileDescriptor fileDescriptor = FileDescriptorUtils
            .protoFileToFileDescriptor(nonPojoExistentSchemaDefinition, "NonExistent.proto",
                Optional.of("foo"));

        //Create a message using above fileDescriptor
        return DynamicMessage.newBuilder(fileDescriptor.findMessageTypeByName("NonExistentSchema")).build();
    }

    private static final String NAME = "Foo";
    public static final Basic.Customer
        BASIC_REFERENCING_MESSAGE = Basic.Customer.newBuilder().setName(NAME).build();

    public static final DynamicMessage
        BASIC_REFERENCING_DYNAMIC_MESSAGE = DynamicMessage.newBuilder(Basic.Address.getDescriptor())
        .setField(Basic.Address.getDescriptor().findFieldByName("street"), NAME).build();

    public static final BasicSyntax2.Phone
        BASIC_SYNTAX2_MESSAGE = BasicSyntax2.Phone.newBuilder().setModel(NAME).build();

    public static final Basicsyntax3.Phone
        BASIC_SYNTAX3_MESSAGE = Basicsyntax3.Phone.newBuilder().setModel(NAME).build();

    public static final ComplexNestingSyntax3.A.B.C.X.D.F.M
        NESTING_MESSAGE_PROTO3 = ComplexNestingSyntax3.A.B.C.X.D.F.M.newBuilder().setChoice(
        ComplexNestingSyntax3.A.B.C.X.D.F.M.K.L).build();

    public static final ComplexNestingSyntax2.O.A
        NESTING_MESSAGE_PROTO2 = ComplexNestingSyntax2.O.A.newBuilder().addB("12312").build();

    public static final A.B.C.X.D.F.M NESTING_MESSAGE_PROTO3_MULTIPLE_FILES =
        A.B.C.X.D.F.M.newBuilder().setChoice(
            A.B.C.X.D.F.M.K.L).build();

    public static final com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.multiplefiles.Phone
        JAVA_OUTER_CLASS_WITH_MULTIPLE_FILES_MESSAGE =
        com.amazonaws.services.schemaregistry.tests.protobuf.syntax3.multiplefiles.Phone.newBuilder().build();

    public static final Contact.Phone JAVA_OUTER_CLASS_MESSAGE = Contact.Phone.newBuilder().build();

    public static final SnakeCaseFile.snake_case_message SNAKE_CASE_MESSAGE =
        SnakeCaseFile.snake_case_message.newBuilder().build();

    public static final AnotherSnakeCaseProtoFile.another_SnakeCase_
        ANOTHER_SNAKE_CASE_MESSAGE = AnotherSnakeCaseProtoFile.another_SnakeCase_.newBuilder().build();

    public static final Foo1.Dollar DOLLAR_SYNTAX_3_MESSAGE = Foo1.Dollar.newBuilder().build();
    public static final HyphenAtedProtoFile.hyphenated HYPHEN_ATED_PROTO_FILE_MESSAGE =
        HyphenAtedProtoFile.hyphenated.newBuilder().build();

    public static final ProtodevelaslProtoProtoProtodevelBar3.bar DOUBLE_PROTO_WITH_TRAILING_HASH_MESSAGE =
        ProtodevelaslProtoProtoProtodevelBar3.bar.newBuilder().build();

    public static final Unicode.uni UNICODE_MESSAGE =
        Unicode.uni.newBuilder().build();

    public static final ConflictingNameOuterClass.ConflictingName CONFLICTING_NAME_MESSAGE =
        ConflictingNameOuterClass.ConflictingName.newBuilder().build();

    public static final NestedConflictingClassNameOuterClass.Parent.NestedConflictingClassName NESTED_CONFLICTING_NAME_MESSAGE =
        NestedConflictingClassNameOuterClass.Parent.NestedConflictingClassName.newBuilder().build();

    public static final AllTypes ALL_TYPES_MESSAGE_SYNTAX3 =
        AllTypes.newBuilder()
            .setStringType("0asd29340932")
            .setByteType(ByteString.copyFrom(UNICODE_MESSAGE.toByteArray()))
            .setOneOfInt(93)
            .setOneOfMoney(Money.newBuilder().setCurrencyCode("INR").setUnits(4l).setNanos(2390).build())
            .addAllRepeatedString(ImmutableList.of("asd", "fgf"))
            .addAllRepeatedPackedInts(ImmutableList.of("1", "90", "34"))
            .setAnotherOneOfMoney(Money.newBuilder().setCurrencyCode("INR").setUnits(4l).setNanos(2390).build())
            .setOptionalSfixed32(1231)
            .setOptionalSfixed64(3092l)
            .setAnEnum2(AnEnum.ALPHA)
            .setUint64Type(1922l)
            .setInt32Type(91)
            .setSint32Type(-910)
            .setSint64Type(-9122)
            .setFixed32Type(19023)
            .setFixed64Type(123)
            .setNestedMessage1(AllTypes.NestedMessage1.newBuilder().setDoubleType(123123.1232).build())
            .putAComplexMap(90, AnotherTopLevelMessage.NestedMessage2.newBuilder().addAllATimestamp(
                ImmutableList.of(
                    Timestamp.newBuilder().setSeconds(123).setNanos(1).build(),
                    Timestamp.newBuilder().setNanos(0).build()
                )
            ).build())
            .setAnEnum1(AnEnum.BETA)
            .putAComplexMap(81, AnotherTopLevelMessage.NestedMessage2.newBuilder().addATimestamp(Timestamp.newBuilder().build()).build())
            .build();

    public static final AllTypesSyntax2.AllTypes ALL_TYPES_MESSAGE_SYNTAX2 =
        AllTypesSyntax2.AllTypes.newBuilder()
            .setStringType("0asd29340932")
            .setByteType(ByteString.copyFrom(UNICODE_MESSAGE.toByteArray()))
            .setOneOfInt(93)
            .setOneOfMoney(Money.newBuilder().setCurrencyCode("INR").setUnits(4l).setNanos(2390).build())
            .addAllRepeatedString(ImmutableList.of("asd", "fgf"))
            .addAllRepeatedPackedInts(ImmutableList.of("1", "90", "34"))
            .setAnotherOneOfMoney(Money.newBuilder().setCurrencyCode("INR").setUnits(4l).setNanos(2390).build())
            .setOptionalSfixed32(1231)
            .setOptionalSfixed64(3092l)
            .setAnEnum2(AllTypesSyntax2.AnEnum.BETA)
            .setUint64Type(1922l)
            .setInt32Type(91)
            .setSint32Type(-910)
            .setSint64Type(-9122)
            .setFixed32Type(19023)
            .setFixed64Type(123)
            .setNestedMessage1(AllTypesSyntax2.AllTypes.NestedMessage1.newBuilder().setDoubleType(123123.1232).build())
            .putAComplexMap(90,
                AllTypesSyntax2.AnotherTopLevelMessage.NestedMessage2.newBuilder()
                    .addAllATimestamp(
                    ImmutableList.of(
                        Timestamp.newBuilder().setSeconds(123).setNanos(1).build(),
                        Timestamp.newBuilder().setNanos(0).build()
                    )
                ).build())
            .build();

    public static final WellKnownTypesTestSyntax2.WellKnownTypesSyntax3 WELL_KNOWN_TYPES_SYNTAX_2 =
        WellKnownTypesTestSyntax2.WellKnownTypesSyntax3.newBuilder()
            .setA(101)
            .setFloating(0)
            .setF1(Timestamp.newBuilder().setSeconds(123).setNanos(1).build())
            .setF2(StringValue.newBuilder().setValue("stringValue").build())
            .setF4(Empty.newBuilder().build())
            .setF5(Duration.newBuilder().setNanos(5).setSeconds(10).build())
            .setF22(ListValue.newBuilder().addValues(Value.newBuilder().setNumberValue(2.2).build()).build())
            .setF27(Int32Value.newBuilder().setValue(27).build())
            .setF33(Struct.newBuilder().build())
            .setF35(UInt64Value.newBuilder().setValue(64).build())
            .setF37(Api.newBuilder().setName("newapi").build())
            .setF42(Enum.newBuilder().addEnumvalue(EnumValue.newBuilder().setName("enumValue").build()).build())
            .setF47(Method.newBuilder().setName("method").setRequestTypeUrl("sampleUrl").build())
            .setF48(Mixin.newBuilder().setName("mixin").build())
            .setF9(CalendarPeriod.DAY)
            .setF10(Color.newBuilder().setRed(100).setGreen(100).setBlue(100).build())
            .setF7(Date.newBuilder().setDay(1).setMonth(4).setYear(2022).build())
            .setF13(Fraction.newBuilder().setDenominator(100).setNumerator(9).build())
            .setF6(Money.newBuilder().setUnits(10).build())
            .setF14(Month.APRIL)
            .setF16(PostalAddress.newBuilder().setPostalCode("98121").build())
            .setF15(PhoneNumber.newBuilder().setE164Number("206").build())
            .build();

    public static final WellKnownTypesTestSyntax3.WellKnownTypesSyntax3 WELL_KNOWN_TYPES_SYNTAX_3 =
        WellKnownTypesTestSyntax3.WellKnownTypesSyntax3.newBuilder()
            .setA(101)
            .setFloating(0)
            .setF1(Timestamp.newBuilder().setSeconds(123).setNanos(1).build())
            .setF2(StringValue.newBuilder().setValue("stringValue").build())
            .setF4(Empty.newBuilder().build())
            .setF5(Duration.newBuilder().setNanos(5).setSeconds(10).build())
            .setF22(ListValue.newBuilder().addValues(Value.newBuilder().setNumberValue(2.2).build()).build())
            .setF27(Int32Value.newBuilder().setValue(27).build())
            .setF33(Struct.newBuilder().build())
            .setF35(UInt64Value.newBuilder().setValue(64).build())
            .setF37(Api.newBuilder().setName("newapi").build())
            .setF42(Enum.newBuilder().addEnumvalue(EnumValue.newBuilder().setName("enumValue").build()).build())
            .setF47(Method.newBuilder().setName("method").setRequestTypeUrl("sampleUrl").build())
            .setF48(Mixin.newBuilder().setName("mixin").build())
            .setF9(CalendarPeriod.DAY)
            .setF10(Color.newBuilder().setRed(100).setGreen(100).setBlue(100).build())
            .setF7(Date.newBuilder().setDay(1).setMonth(4).setYear(2022).build())
            .setF13(Fraction.newBuilder().setDenominator(100).setNumerator(9).build())
            .setF6(Money.newBuilder().setUnits(10).build())
            .setF14(Month.APRIL)
            .setF16(PostalAddress.newBuilder().setPostalCode("98121").build())
            .setF15(PhoneNumber.newBuilder().setE164Number("206").build())
            .build();
}

File: native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe.Tests/utils/ProtobufGenerator.cs
// Copyright 2020 Amazon.com, Inc. or its affiliates.
// Licensed under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//  
//     http://www.apache.org/licenses/LICENSE-2.0
//  
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

using Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax3.Alltypes;
using Google.Protobuf;
using Google.Protobuf.WellKnownTypes;
using Google.Type;
using A = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax3.Multiplefiles.A;
using AllTypesSyntax2 = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax2.Alltypes;
using AnotherSnakeCaseProtoFile = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax3.SnakeCase;
using Basicsyntax2 = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax2.Basic;
using Basicsyntax3 = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax3.Basic;
using ComplexNestingSyntax2 = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax2;
using ComplexNestingSyntax3 = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax3;
using Multiplefiles = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax3.Multiplefiles;
using SnakeCaseFileSyntax2 = Com.Amazonaws.Services.Schemaregistry.Tests.Protobuf.Syntax2.SnakeCase;
using WellKnownTypesTestSyntax2 = Com.Amazonaws.Services.Schemaregistry.Utils.Apicurio.Syntax2;
using WellKnownTypesTestSyntax3 = Com.Amazonaws.Services.Schemaregistry.Utils.Apicurio.Syntax3;

// ReSharper disable InconsistentNaming
namespace AWSGsrSerDe.Tests.utils
{
    public static class ProtobufGenerator
    {
        private const string Name = "Customer Name";

        public static readonly ComplexNestingSyntax2.Customer BASIC_REFERENCING_MESSAGE =
            new ComplexNestingSyntax2.Customer
            {
                Name = Name,
            };

        public static readonly Basicsyntax2.Phone BASIC_SYNTAX2_MESSAGE = new Basicsyntax2.Phone
        {
            Name = Name,
        };

        public static readonly Basicsyntax3.Phone BASIC_SYNTAX3_MESSAGE = new Basicsyntax3.Phone
        {
            Name = Name,
        };

        public static readonly ComplexNestingSyntax3.A.Types.B.Types.C.Types.X.Types.D.Types.F.Types.M
            NESTING_MESSAGE_PROTO3 =
                new ComplexNestingSyntax3.A.Types.B.Types.C.Types.X.Types.D.Types.F.Types.M
                {
                    Choice = ComplexNestingSyntax3.A.Types.B.Types.C.Types.X.Types.D.Types.F.Types.M.Types.K.L,
                };

        public static readonly ComplexNestingSyntax2.O.Types.A NESTING_MESSAGE_PROTO2 =
            new ComplexNestingSyntax2.O.Types.A
            {
                B = { "12312", },
            };

        public static readonly A.Types.B.Types.C.Types.X.Types.D.Types.F.Types.M NESTING_MESSAGE_PROTO3_MULTIPLE_FILES =
            new A.Types.B.Types.C.Types.X.Types.D.Types.F.Types.M
            {
                Choice = A.Types.B.Types.C.Types.X.Types.D.Types.F.Types.M.Types.K.L,
            };

        public static readonly Multiplefiles.Phone DOTNET_OUTER_CLASS_WITH_MULTIPLE_FILES_MESSAGE =
            new Multiplefiles.Phone();

        public static readonly SnakeCaseFileSyntax2.snake_case_message SNAKE_CASE_MESSAGE =
            new SnakeCaseFileSyntax2.snake_case_message();

        public static readonly AnotherSnakeCaseProtoFile.another_SnakeCase_ ANOTHER_SNAKE_CASE_MESSAGE =
            new AnotherSnakeCaseProtoFile.another_SnakeCase_();

        public static readonly Basicsyntax3.Dollar DOLLAR_SYNTAX_3_MESSAGE = new Basicsyntax3.Dollar();

        public static readonly Basicsyntax3.hyphenated HYPHEN_ATED_PROTO_FILE_MESSAGE = new Basicsyntax3.hyphenated();

        public static readonly Basicsyntax2.bar DOUBLE_PROTO_WITH_TRAILING_HASH_MESSAGE = new Basicsyntax2.bar();

        public static readonly Basicsyntax3.uni UNICODE_MESSAGE = new Basicsyntax3.uni();

        public static readonly Basicsyntax3.ConflictingName CONFLICTING_NAME_MESSAGE =
            new Basicsyntax3.ConflictingName();

        public static readonly Basicsyntax3.Parent.Types.NestedConflictingClassName NESTED_CONFLICTING_NAME_MESSAGE =
            new Basicsyntax3.Parent.Types.NestedConflictingClassName();

        public static readonly AllTypes ALL_TYPES_MESSAGE_SYNTAX3 = new AllTypes
        {
            StringType = "0asd29340932",
            ByteType = ByteString.CopyFrom(UNICODE_MESSAGE.ToByteArray()),
            OneOfInt = 93,
            OneOfMoney = new Money
            {
                CurrencyCode = "INR",
                Units = 4L,
                Nanos = 2390,
            },
            RepeatedString = { "asd", "fgf" },
            RepeatedPackedInts = { "1", "90", "34" },
            AnotherOneOfMoney = new Money
            {
                CurrencyCode = "INR",
                Units = 4L,
                Nanos = 2390,
            },
            OptionalSfixed32 = 1231,
            OptionalSfixed64 = 3092L,
            AnEnum2 = AnEnum.Alpha,
            Uint64Type = 1922L,
            Int32Type = 91,
            Sint32Type = -910,
            Sint64Type = -9122,
            Fixed32Type = 19023,
            Fixed64Type = 123,
            NestedMessage1 = new AllTypes.Types.NestedMessage1 { DoubleType = 123123.1232 },
            AComplexMap =
            {
                {
                    90, new AnotherTopLevelMessage.Types.NestedMessage2
                    {
                        ATimestamp =
                        {
                            new Timestamp
                            {
                                Seconds = 123,
                                Nanos = 1,
                            },
                            new Timestamp
                            {
                                Nanos = 0,
                            },
                        },
                    }
                },
                {
                    81, new AnotherTopLevelMessage.Types.NestedMessage2
                    {
                        ATimestamp = { new Timestamp() },
                    }
                },
            },
            AnEnum1 = AnEnum.Beta,
        };

        public static readonly AllTypesSyntax2.AllTypes ALL_TYPES_MESSAGE_SYNTAX2 = new AllTypesSyntax2.AllTypes
        {
            StringType = "0asd29340932",
            ByteType = ByteString.CopyFrom(UNICODE_MESSAGE.ToByteArray()),
            OneOfInt = 93,
            OneOfMoney = new Money
            {
                CurrencyCode = "INR",
                Units = 4L,
                Nanos = 2390,
            },
            RepeatedString = { "asd", "fgf" },
            RepeatedPackedInts = { "1", "90", "34" },
            AnotherOneOfMoney = new Money
            {
                CurrencyCode = "INR",
                Units = 4L,
                Nanos = 2390,
            },
            OptionalSfixed32 = 1231,
            OptionalSfixed64 = 3092L,
            AnEnum2 = AllTypesSyntax2.AnEnum.Beta,
            Uint64Type = 1922L,
            Int32Type = 91,
            Sint32Type = -910,
            Sint64Type = -9122,
            Fixed32Type = 19023,
            Fixed64Type = 123,
            NestedMessage1 = new AllTypesSyntax2.AllTypes.Types.NestedMessage1() { DoubleType = 123123.1232 },
            AComplexMap =
            {
                {
                    90, new AllTypesSyntax2.AnotherTopLevelMessage.Types.NestedMessage2()
                    {
                        ATimestamp =
                        {
                            new Timestamp
                            {
                                Seconds = 123,
                                Nanos = 1,
                            },
                            new Timestamp
                            {
                                Nanos = 0,
                            },
                        },
                    }
                },
            },
        };

        public static readonly WellKnownTypesTestSyntax2.WellKnownTypesSyntax3 WELL_KNOWN_TYPES_SYNTAX_2 =
            new WellKnownTypesTestSyntax2.WellKnownTypesSyntax3
            {
                A = 101,
                Floating = 0,
                F1 = new Timestamp { Seconds = 123, Nanos = 1 },
                F2 = "stringValue",
                F4 = new Empty(),
                F5 = new Duration { Nanos = 5, Seconds = 10 },
                F22 = new ListValue { Values = { new Value { NumberValue = 2.2 } } },
                F27 = 27,
                F33 = new Struct(),
                F35 = 64L,
                F37 = new Api { Name = "newapi" },
                F42 = new Enum { Enumvalue = { new EnumValue { Name = "enumValue" } } },
                F47 = new Method { Name = "method", RequestTypeUrl = "sampleUrl" },
                F48 = new Mixin { Name = "mixin" },
                F9 = CalendarPeriod.Day,
                F10 = new Color { Red = 100, Green = 100, Blue = 1000 },
                F7 = new Date { Day = 1, Month = 4, Year = 2022 },
                F13 = new Fraction { Denominator = 100, Numerator = 9 },
                F6 = new Money { Units = 10 },
                F14 = Month.April,
                F16 = new PostalAddress { PostalCode = "98121" },
                F15 = new PhoneNumber { E164Number = "206" },
            };

        public static readonly WellKnownTypesTestSyntax3.WellKnownTypesSyntax3 WELL_KNOWN_TYPES_SYNTAX_3 =
            new WellKnownTypesTestSyntax3.WellKnownTypesSyntax3
            {
                A = 101,
                Floating = 0,
                F1 = new Timestamp { Seconds = 123, Nanos = 1 },
                F2 = "stringValue",
                F4 = new Empty(),
                F5 = new Duration { Nanos = 5, Seconds = 10 },
                F22 = new ListValue { Values = { new Value { NumberValue = 2.2 } } },
                F27 = 27,
                F33 = new Struct(),
                F35 = 64L,
                F37 = new Api { Name = "newapi" },
                F42 = new Enum { Enumvalue = { new EnumValue { Name = "enumValue" } } },
                F47 = new Method { Name = "method", RequestTypeUrl = "sampleUrl" },
                F48 = new Mixin { Name = "mixin" },
                F9 = CalendarPeriod.Day,
                F10 = new Color { Red = 100, Green = 100, Blue = 1000 },
                F7 = new Date { Day = 1, Month = 4, Year = 2022 },
                F13 = new Fraction { Denominator = 100, Numerator = 9 },
                F6 = new Money { Units = 10 },
                F14 = Month.April,
                F16 = new PostalAddress { PostalCode = "98121" },
                F15 = new PhoneNumber { E164Number = "206" },
            };
    }
}
